extensions/common/aws/aws-s3-test/src/testFixtures/java/org/eclipse/edc/aws/s3/testfixtures/AbstractS3Test.java

```diff
@@ -66,147 +66,183 @@ public abstract class AbstractS3Test {
     // when bucket is rapidly added/deleted and consistency propagation causes this error.
     // (Should not be necessary if REGION remains static, but added to prevent future frustration.)
     // [see http://stackoverflow.com/questions/13898057/aws-error-message-a-conflicting-conditional-operation-is-currently-in-progress]
-    protected static final String MINIO_ENDPOINT = "http://localhost:9000";
-    protected static final URI S3_ENDPOINT = URI.create(propOrEnv("it.aws.endpoint", MINIO_ENDPOINT));
+    protected static final String SOURCE_MINIO_ENDPOINT = "http://localhost:9000";
```

Comment:
I see a lot of duplication added in this class, a different way of refactor this could be to have a class representing Minio (offering methods to create buckets, create clients and so on), with two instances, one for `source` and one for `provider`. It will simplify code a little

Reply:
Sounds right. I see there is an official Minio java client: https://min.io/docs/minio/linux/developers/java/minio-java.html, but I have no clue about how to include it into the codebase - does it need to be approved? Where can I find some information about the process of adding new deps? Thanks in advance!

Reply:
I don't think we need a client here, we are using MinIO just for testing purpose but the extensions are meant to work with AWS S3.
I was just talking about a refactoring, the logic is already implemented and works, it's just a matter of tiding up things a litte.